### PR TITLE
Add missing VUID-vkCmdExecuteCommand-pCommandBuffers checks and tests

### PR DESCRIPTION
**Pull request by r-potter**
MR is complete and can be reviewed, but will need an update when https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/8298 lands

adds

- VUID-vkCmdExecuteCommands-pCommandBuffers-09504
- VUID-vkCmdExecuteCommands-pCommandBuffers-09505

### CODE REVIEW

File: layers/core_checks/cc_cmd_buffer.cpp [hunk 1]
```diff
@@ -879,6 +879,91 @@ bool CoreChecks::PreCallValidateCmdExecuteCommands(VkCommandBuffer commandBuffer
...
+                        sub_cb_state.activeRenderPass->inheritance_rendering_info.pNext);
+
+                    if (location_info) {
+                        const std::string vuid_090504 = "VUID-vkCmdExecuteCommands-pCommandBuffers-09504";
+                        const LogObjectList objlist(commandBuffer, pCommandBuffers[i]);
+                        skip |= ValidateRenderingAttachmentLocationsKHR(*location_info, objlist, cb_loc.dot(Field::pNext));
+
+                        if (location_info->colorAttachmentCount != cb_state.rendering_attachments.color_indexes.size()) {
+                            skip |=
+                                LogError(vuid_090504, objlist, cb_loc.dot(Field::pNext),
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_cmd_buffer.cpp:895]:
```suggestion
                                LogError(vuid_090504, objlist, cb_loc.pNext(Struct::VkRenderingAttachmentLocationInfoKHR, Field::colorAttachmentCount), "(%" PRIu32 ") does not match the implicit ....
```

will print out a nicer error message showing where this occured

File: layers/core_checks/cc_cmd_buffer.cpp [hunk 2]
```diff
@@ -879,6 +879,91 @@ bool CoreChecks::PreCallValidateCmdExecuteCommands(VkCommandBuffer commandBuffer
...
+                                         "VkRenderingAttachmentLocationInfoKHR::colorAttachmentCount = %" PRIu32
+                                         " does not match the implicit or explicit state in the primary command buffer, "
+                                         "colorAttachmentCount = %" PRIu64 ".",
+                                         location_info->colorAttachmentCount, cb_state.rendering_attachments.color_indexes.size());
+                        } else {
+                            for (uint32_t idx = 0; idx < location_info->colorAttachmentCount; idx++) {
+                                if (location_info->pColorAttachmentLocations &&
+                                    location_info->pColorAttachmentLocations[idx] !=
+                                        cb_state.rendering_attachments.color_locations[idx]) {
+                                    skip |= LogError(vuid_090504, objlist, cb_loc.dot(Field::pNext),
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_cmd_buffer.cpp:905]: `cb_loc.pNext(Struct::VkRenderingAttachmentLocationInfoKHR, Field::pColorAttachmentsLocations, idx)`

File: layers/core_checks/cc_cmd_buffer.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/core_checks/cc_cmd_buffer.cpp:895]: same for all checks below

**Comment by r-potter** [comment]
Thanks @spencer-lunarg. I'll get these fixed up, plus the typo tweaks

File: layers/core_checks/cc_cmd_buffer.cpp [hunk 3]
```diff
@@ -879,6 +880,98 @@ bool CoreChecks::PreCallValidateCmdExecuteCommands(VkCommandBuffer commandBuffer
...
+                                "(%" PRIu32
+                                ") does not match the implicit or explicit state in the primary command buffer ("
+                                "%" PRIu32 ").",
+                                location_info->colorAttachmentCount, unsigned(cb_state.rendering_attachments.color_indexes.size()));
+                        } else {
+                            for (uint32_t idx = 0; idx < location_info->colorAttachmentCount; idx++) {
+                                if (location_info->pColorAttachmentLocations &&
+                                    location_info->pColorAttachmentLocations[idx] !=
+                                        cb_state.rendering_attachments.color_locations[idx]) {
+                                    skip |= LogError(vuid_090504, objlist, cb_loc.dot(Field::pNext),
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_cmd_buffer.cpp:907]: you miss this one? (rest look good)

**Review by spencer-lunarg — APPROVED** [review]

File: layers/core_checks/cc_cmd_buffer.cpp (hunk 3 above)
Comment (r-potter) [reply layers/core_checks/cc_cmd_buffer.cpp:907]: Of course I did 🤦 Thanks Spencer

**Pull request merged**